### PR TITLE
docs(HTTP_API): document JWT asset download endpoints

### LINKED ISSUE #995: [Task] Document JWT-consuming V2 API handlers
When do I know if I should update the v1 vs. v2 handlers or both? In my mind, this is a newer feature so it would make more sense to just update the v2 ReportGetHandler and leave the v1 version as an older version however I notice that that v2 version is not documented in the HTTP API yet.

_Originally posted by @maxcao13 in https://github.com/cryostatio/cryostat/issues/880#issuecomment-1152753428_

### PR DESCRIPTION
**Pull request by andrewazores**
- docs(HTTP_API): update GraphQL Authorization description
- docs(HTTP_API): document JWT asset download handlers

Fixes #995

### CODE REVIEW

File: docs/HTTP_API.md [hunk 1]
```diff
@@ -1705,6 +1791,91 @@ The handler-specific descriptions below describe how each handler populates the
     {"meta":{"status":"OK","type":"application/json"},"data":{result:[{"name":"Name","description":"Recording name","defaultValue":"Recording"},{"name":"Duration","description":"Duration of recording","defaultValue":"30s[s]"},{"name":"Max Size","description":"Maximum size of recording","defaultValue":"0B[B]"},{"name":"Max Age","description":"Maximum age of the events in the recording","defaultValue":"0s[s]"},{"name":"To disk","description":"Record to disk","defaultValue":"false"},{"name":"Dump on Exit","description":"Dump recording data to disk on JVM exit","defaultValue":"false"}]}}
     ```
 
+* #### `TargetRecordingGetHandler`
+
+    ###### synopsis
+    Returns a recording of a target JVM as an octet stream, using a JWT for
+    authorization. See `AuthTokenPostHandler`.
```
Comment (tthvo) [inline docs/HTTP_API.md:1798]: I guess we could add a quick link back to `AuthTokenPostHandler` section? Same for others that have `See [handler]` clause?

File: docs/HTTP_API.md [hunk 2]
```diff
@@ -1705,6 +1791,91 @@ The handler-specific descriptions below describe how each handler populates the
...
+    Should use percent-encoding.
+
+    `recordingName` - The name of the recording to get the report for.
+    Should use percent-encoding.
+
+    `jwt` - The JSON Web Token providing authorization for this request. See
+    `AuthTokenPostHandler`.
+
+    ###### response
+    `200` - The body is the requested report as an HTML document.
```
Comment (tthvo) [inline docs/HTTP_API.md:1856]: I remember you mentioned about generating in either `.html` or `.json`. Any reference on that feat here?

File: docs/HTTP_API.md [hunk 3]
```diff
@@ -1748,6 +1919,67 @@ The handler-specific descriptions below describe how each handler populates the
     {"meta":{"status":"Created","type":"application/json"},"data":{"result":{"downloadUrl":"http://192.168.0.109:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:9091%2Fjmxrmi/recordings/snapshot-1","reportUrl":"http://192.168.0.109:8181/api/v1/targets/service:jmx:rmi:%2F%2F%2Fjndi%2Frmi:%2F%2Flocalhost:9091%2Fjmxrmi/reports/snapshot-1","id":1,"name":"snapshot-1","state":"STOPPED","startTime":1601998841300,"duration":0,"continuous":true,"toDisk":true,"maxSize":0,"maxAge":0}}}
     ```
 
+### Recordings in Archive
```
Comment (tthvo) [inline docs/HTTP_API.md:1922]: Target-specific handlers will later replace these handlers and deprecate them, I think. So, will these be marked depr or removed then?

File: docs/HTTP_API.md (hunk 2 above)
Comment (andrewazores) [reply docs/HTTP_API.md:1856]: https://github.com/cryostatio/cryostat/pull/1009

Comment (andrewazores) [reply docs/HTTP_API.md:1856]: The reports sidecar is able to do that already, but this endpoint on the main Cryostat API gateway doesn't support it (yet). Needs #1009 which needs #1072 first.

Comment (tthvo) [reply docs/HTTP_API.md:1874]: Right! Docs can be updated after those then!

**Review by maxcao13 — COMMENTED** [review]
Should the web-client be updated in this PR?

**Comment by andrewazores** [comment]
> Should the web-client be updated in this PR?

Thanks for catching that, not sure what happened there.

**Review by maxcao13 — DISMISSED** [review]
Looks good to me now!

**Review by tthvo — APPROVED** [review]
Looks good to me too!

**Review by tthvo — APPROVED** [review]

**Pull request merged**